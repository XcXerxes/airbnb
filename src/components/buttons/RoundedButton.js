import React, {Component} from 'react'
import {View, Text, TouchableOpacity, Image, StyleSheet} from 'react-native'
import PropTypes from 'prop-types'
import colors from '../../styles/colors'

export default class RoundedButton extends Component {
  render () {
    const {text, 
      background,
      loading,
      disabled,
      textSize,
      textWeight,
      iconPosition,
      textAlign,
      borderColor,
      textColor,
      handleOnPress,
      icon} = this.props
    const color = textColor || colors.black
    const backgroundColor = background || 'transparent'
    const fontSize = textSize || 16
    const fontWeight = textWeight || '600'
    const alignPosition = textAlign || 'center'
    const iconLocation = iconPosition || 'left'
    const border = borderColor || color.white
    const opacityStyle = disabled || loading ? .5 : 1
    const textOpacity = loading ? 0 : 1

    return (
      <TouchableOpacity 
        onPress={handleOnPress}
        style={[styles.wrapper, {opacity: opacityStyle, borderColor: border, backgroundColor}]}
        activeOpacity={0.7}
        disabled={disabled || loading}
      >
        <View style={styles.buttonTextWrapper}
        
        >
          {iconLocation === 'left' && !loading ? icon : null}
          {loading ?
           <View style={styles.loaderContainer}>
             <Image style={styles.loaderImage} 
              source={require('../../img/loading.gif')}
             />
           </View> : null
          }
          <Text 
          style={[styles.buttonText, {opacity: opacityStyle, color, fontSize, fontWeight, textAlign, alignPosition}]}
          >{text}</Text>
          {iconLocation === 'right' && !loading ? icon : null}
        </View>
      </TouchableOpacity>
    )
  }
}

RoundedButton.propTypes = {
  text: PropTypes.string.isRequired,
  background: PropTypes.string,
  textColor: PropTypes.string,
  handleOnPress: PropTypes.func,
  icon: PropTypes.object,
  textSize: PropTypes.string,
  textWeight: PropTypes.string,
  textAlign: PropTypes.string,
  borderColor: PropTypes.string,
  iconPosition: PropTypes.string,
  disabled: PropTypes.bool,
  loading: PropTypes.bool
}

const styles = StyleSheet.create({
  wrapper: {
    display: 'flex',
    paddingLeft: 20,
    paddingRight: 20,
    paddingTop: 12,
    paddingBottom: 12,
    borderRadius: 40,
    borderWidth: 1,
    marginBottom: 15,
    alignItems: 'center'
    
  },
  buttonTextWrapper: {
    flexDirection: 'row',
    justifyContent: 'flex-end'
  },
  buttonText: {
    width: '100%'
  },
  loaderContainer: {
    width: 90,
    height: 90,
    borderRadius: 15,
    position: 'absolute',
    left: '50%',
    marginLeft: -45,
    top: '50%',
    marginTop: -45
  },
  loaderImage: {
    position: 'absolute',
    width: 40,
    height: 40,
    left: '50%',
    marginLeft: -20,
    top: '50%',
    marginTop: -20,
    borderRadius: 15
  }
})