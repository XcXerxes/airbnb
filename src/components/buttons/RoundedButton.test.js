import React from 'react'
import RoundedButton from './RoundedButton'

import render from 'react-test-renderer'

const treeRender = render.create(<RoundedButton text="测试" textColor="#000000" />)
const tree = treeRender.toJSON()
const treeRoot = treeRender.root
console.log(treeRoot.props)
it('renders correctly shot', () => {
  expect(tree).toMatchSnapshot()
  // expect(treeRoot.findByType(RoundedButton).props.background).toBe('transparent')
})

it('should renders props of text values', () => {
  expect(treeRoot.findByType(RoundedButton).props.text).toBe('测试')
})

it('should renders props of background', () => {
  expect(treeRoot.findByType(RoundedButton).props.background).toBe(undefined)
})

it('should renders props of textColor', () => {
  expect(treeRoot.findByType(RoundedButton).props.textColor).toBe("#000000")
})

it('should renders props numbers', () => {
  expect(Object.keys(treeRoot.props).length).toBe(2)
})
