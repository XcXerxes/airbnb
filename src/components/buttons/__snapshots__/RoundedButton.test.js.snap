// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly shot 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderColor": "#ffffff",
        "borderRadius": 40,
        "borderWidth": 1,
        "display": "flex",
        "marginBottom": 15,
        "padding": 15,
      },
      Object {
        "backgroundColor": "transparent",
      },
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "flex-end",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 16,
            "textAlign": "center",
            "width": "100%",
          },
          Object {
            "color": "#000000",
          },
        ]
      }
    >
      测试
    </Text>
  </View>
</View>
`;
